rewrite-java-test/src/test/java/org/openrewrite/java/MethodMatcherTest.java

```diff
@@ -140,15 +140,49 @@ void matchesUnqualifiedJavaLangArguments() {
     void matchesArgumentsWithWildcards() {
         assertTrue(argRegex("A foo(java.util.*)").matcher("java.util.Map").matches());
         assertTrue(argRegex("A foo(java..*)").matcher("java.util.Map").matches());
+        assertTrue(argRegex("A foo(*.util.*)").matcher("java.util.Map").matches());
+        assertTrue(argRegex("A foo(*..*)").matcher("java.util.Map").matches());
+    }
+
+    @Test
+    void matchesExactlyOneWithWildcard() {
+        assertTrue(argRegex("A foo(*)").matcher("int").matches());
+        assertTrue(argRegex("A foo(*, int)").matcher("int,int").matches());
+        assertTrue(argRegex("A foo(*, int)").matcher("double,int").matches());
+        assertTrue(argRegex("A foo(int, *)").matcher("int,int").matches());
+        assertTrue(argRegex("A foo(int, *)").matcher("int,double").matches());
+        assertTrue(argRegex("A foo(*, *)").matcher("int,int").matches());
+        assertTrue(argRegex("A foo(int, *, double)").matcher("int,int,double").matches());
+        assertTrue(argRegex("A foo(int, *, double)").matcher("int,double,double").matches());
+
+        assertFalse(argRegex("A foo(*)").matcher("").matches());
+        assertFalse(argRegex("A foo(*)").matcher("int,int").matches());
+        assertFalse(argRegex("A foo(*, int)").matcher("int").matches());
+        assertFalse(argRegex("A foo(*, int)").matcher("int,double").matches());
+        assertFalse(argRegex("A foo(int, *)").matcher("int").matches());
+        assertFalse(argRegex("A foo(int, *)").matcher("double,int").matches());
+        assertFalse(argRegex("A foo(*, *)").matcher("").matches());
+        assertFalse(argRegex("A foo(*, *)").matcher("int").matches());
+        assertFalse(argRegex("A foo(int, *, double)").matcher("int,double").matches());
+        assertFalse(argRegex("A foo(int, *, double)").matcher("double,int,double").matches());
     }
 
     @Test
     void matchesArgumentsWithDotDot() {
         assertTrue(argRegex("A foo(.., int)").matcher("int").matches());
         assertTrue(argRegex("A foo(.., int)").matcher("int,int").matches());
+        assertTrue(argRegex("A foo(.., int)").matcher("double,int").matches());
+        assertFalse(argRegex("A foo(.., int)").matcher("int,double").matches());
```

Comment:
As .. would mean one or more, I guess this works too:

```java
assertTrue(argRegex("A foo(.., int)").matcher("double,double,int").matches());
assertTrue(argRegex("A foo(int, .., double)").matcher("int,double,String,int,double").matches());
```

So add these tests as well!